integration-libs/opf/base/root/services/opf-state-persistence.service.spec.ts

```diff
@@ -1 +1,119 @@
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { TestBed } from '@angular/core/testing';
+import { StatePersistenceService } from '@spartacus/core';
+import { BehaviorSubject, of, Subscription } from 'rxjs';
+import { OpfPaymentMetadata } from '../model';
+import { OpfPaymentMetadataStoreService } from './opf-payment-metadata-store.service';
+import {
+  OpfStatePersistenceService,
+  SyncedOpfState,
+} from './opf-state-persistence.service';
+
+describe('OpfStatePersistenceService', () => {
+  let service: OpfStatePersistenceService;
+  let statePersistenceServiceMock: jasmine.SpyObj<StatePersistenceService>;
+  let opfPaymentMetadataStoreServiceMock: jasmine.SpyObj<OpfPaymentMetadataStoreService>;
+
+  beforeEach(() => {
+    statePersistenceServiceMock = jasmine.createSpyObj(
+      'StatePersistenceService',
+      ['syncWithStorage']
+    );
+    opfPaymentMetadataStoreServiceMock = jasmine.createSpyObj(
+      'OpfPaymentMetadataStoreService',
+      ['getOpfMetadataState', 'updateOpfMetadata']
+    );
+
+    TestBed.configureTestingModule({
+      providers: [
+        OpfStatePersistenceService,
+        {
+          provide: StatePersistenceService,
+          useValue: statePersistenceServiceMock,
+        },
+        {
+          provide: OpfPaymentMetadataStoreService,
+          useValue: opfPaymentMetadataStoreServiceMock,
+        },
+      ],
+    });
+
+    service = TestBed.inject(OpfStatePersistenceService);
+  });
+
+  it('should be created', () => {
+    expect(service).toBeTruthy();
+  });
+
+  it('should initialize the synchronization with state and browser storage', () => {
+    const mockOpfMetadata: OpfPaymentMetadata = {
+      isPaymentInProgress: true,
+      selectedPaymentOptionId: 111,
+      termsAndConditionsChecked: true,
+    };
+
+    const mockSyncedOpfState: SyncedOpfState = {
+      metadata: mockOpfMetadata,
+    };
+
+    const stateObservable = new BehaviorSubject<SyncedOpfState | undefined>(
+      mockSyncedOpfState
+    );
+    opfPaymentMetadataStoreServiceMock.getOpfMetadataState.and.returnValue(
+      of(stateObservable.value?.metadata)
+    );
+
+    service.initSync();
+
+    expect(statePersistenceServiceMock.syncWithStorage).toHaveBeenCalled();
+  });
+
+  it('should get and transform Opf state', (done) => {
+    const mockOpfMetadata: OpfPaymentMetadata = {
+      isPaymentInProgress: true,
+      selectedPaymentOptionId: 111,
+      termsAndConditionsChecked: true,
+    };
+
+    const stateObservable = new BehaviorSubject<OpfPaymentMetadata>(
+      mockOpfMetadata
+    );
+    opfPaymentMetadataStoreServiceMock.getOpfMetadataState.and.returnValue(
+      stateObservable
+    );
+
+    service['getOpfState']().subscribe((state) => {
+      expect(state).toEqual({ metadata: mockOpfMetadata });
+      done();
+    });
+  });
+
+  it('should update OpfPaymentMetadataStoreService when onRead is called', () => {
+    const mockOpfMetadata: OpfPaymentMetadata = {
```

Comment:
Can we declare this const more at the top of the test scenarios? So it's no need to re-declare this property each time.

Reply:
Changed